federation/cluster/common.sh

```diff
@@ -24,7 +24,19 @@
 
 : "${KUBE_ROOT?Must set KUBE_ROOT env var}"
 
+# Provides the $KUBERNETES_PROVIDER variable and detect-project function
+source "${KUBE_ROOT}/cluster/kube-util.sh"
+
+# Populates $PROJECT
+detect-project
+if [[ ${PROJECT} == *':'* ]]
```

Comment:
Also optional: we dont need to run this whole code (including calling detect-project) if FEDERATION_PUSH_REPO_BASE is set already.